importer/main.py

```diff
@@ -1101,6 +1102,58 @@ def export_resources_to_csv(resource_type, parameter, value, limit):
         logging.error(f"Failed to retrieve resource. Status code: {response[1]} response: {response[0]}")
 
 
+def encode_image(image_file):
+    with open(image_file, 'rb') as image:
+        image_b64_data = base64.b64encode(image.read())
+    return image_b64_data
+
+
+# This function takes in the source url of an image, downloads it, encodes it,
+# and saves it as a Binary resource. It returns the id of the Binary resource if
+# successful and 0 if failed
+def save_image(image_source_url):
+    headers = {"Authorization": "Bearer " + config.product_access_token}
+    data = requests.get(url=image_source_url, headers=headers)
+    if data.status_code == 200:
+        with open('images/image_file', 'wb') as image_file:
+            image_file.write(data.content)
+
+        encoded_image = encode_image('images/image_file')
+        resource_id = str(uuid.uuid5(uuid.NAMESPACE_DNS, image_source_url))
+        payload = {
+            "resourceType": "Bundle",
+            "type": "transaction",
+            "entry": [{
+                "request": {
+                    "method": "PUT",
+                    "url": "Binary/" + resource_id,
+                    "ifMatch": "1"
+                },
+                "resource": {
+                    "resourceType": "Binary",
+                    "id": resource_id,
+                    "contentType": "image/png",
```

Comment:
Not sure how we use this contentType here or how important it is... from the current examples available as a source, there is no way to deduce this information, so just have this `png` here as a placeholder for now

Reply:
you should be able to look at the binary date to figure this out, here's first hit, https://stackoverflow.com/questions/43580/how-to-find-the-mime-type-of-a-file-in-python, worth looking a bit more to see if there's something better